pandas/core/generic.py

```diff
@@ -197,6 +202,7 @@ def __init__(
         axes: Optional[List[Index]] = None,
         copy: bool = False,
         dtype: Optional[Dtype] = None,
+        attrs: Optional[Mapping[Hashable, Any]] = None,
```

Comment:
Minor but any reason to prefer `Mapping` over `Dict`? I think should stick with the latter generically

Reply:
This is what xarray does. It's probably best to be forgiving in what we accept. This isn't user-facing (yet), but they could give an OrderedDict say.

Reply:
Dict would encompass OrderedDict since its its parent, so won't make a difference there. There are some edge cases where this would make a difference (i.e. if we called `attrs.copy()` but I think should just stick with `Dict` since we don't have this elsewhere and maybe leave to a separate exercise if we wanted to really fine tune those types

Reply:
I'd prefer to match xarray here.

Reply:
I'm not against it but this isn't the place to start. We don't do this anywhere else.

```sh
$ grep "Dict\[" -r pandas --include="*.py" | wc -l
      60
$ grep "Mapping\[" -r pandas --include="*.py" | wc -l
       0
```

So if you feel strongly about matching xarray should be done in a separate PR comprehensively instead of just one-off here

Reply:
Maybe let's step back: why isn't Mapping the correct type?

Reply:
from https://github.com/python/typeshed/blob/master/CONTRIBUTING.md

Some further tips for good type hints:

avoid invariant collection types (List, Dict) in argument positions, in favor of covariant types like Mapping or Sequence;

Reply:
The storage is a dict, the argument is a mapping.

Reply:
OK. I wasn't expecting this be such a sticking point so whatever everyone else is fine with works

@simonjayhawkins thanks for the link. Interestingly enough there are still more Dict appearances in typeshed than Mapping so guess it doesn't matter too much